### DISCUSSION TITLE
how to debug android webview

### DISCUSSION THREAD

**Discussion by Jzhenli** [beeware/toga#2765] [Q&A] [answered]
Hi, I use toga to build a web view-based GUI backed by a web server for android app, the webview will always shows a white screen, but it's OK to open with browser. I don't know how to debug it in the toga, would u help me? Thanks.
 
 Attached is the example code, web is build with Vue, and server is the python default http.server.
[positron.zip](https://github.com/user-attachments/files/16596088/positron.zip)

**Comment by rmartin16** [comment]
When I run a WebView in an Android emulator, I can remotely access its DevTools via chrome://inspect/ on the host machine. Same can be done for a physical device if USB Debugging, etc. is enabled on the device.

**Reply by rmartin16** [reply]
FWIW, I was able to get your Vue app to at least load once I enabled DOM storage for the WebView via:
```diff
--- a/android/src/toga_android/widgets/webview.py
+++ b/android/src/toga_android/widgets/webview.py
@@ -38,6 +38,8 @@ class WebView(Widget):
         self.settings.setBuiltInZoomControls(True)
         self.settings.setDisplayZoomControls(False)
 
+        self.settings.setDomStorageEnabled(True)
+
     def get_url(self):
         url = self.native.getUrl()
         if url == "about:blank" or url.startswith("data:"):
```
~~I will have to defer to others here with more knowledge about the best way to handle this but I was able to "hacK" this in with the subclass below. (Note: this WebView subclass can only be used on Android.)~~

[EDIT] remove faulty workaround

**Reply by rmartin16** [reply, marked as answer]
Actually, apologies, I misled myself that my proposed workaround (now removed) was working.

Instead, after you create your `WebView`, run this to enable DOM storage for the Android WebView:
```python
if toga.platform.current_platform == "android":
    self.web_view._impl.settings.setDomStorageEnabled(True)
```

**Reply by freakboy3742** [reply]
That seems like an option we should probably enable - I can't think of any reason we should disable DOM storage by default.

**Reply by freakboy3742** [reply]
I've opened #2767 to track this.
